autodocodec/src/Autodocodec/Codec.hs

```diff
@@ -258,6 +283,30 @@ data Union
 
 instance Validity Union
 
+-- | Discriminator value used in 'DiscriminatedUnionCodec'
+type Discriminator = Text
+
+-- | Takes a value and and 'ObjectCodec' for encoding that value
+-- and wraps it in an existential type wrapper.
+-- Used for constructing input encoders for 'DiscriminatedUnionCodec'
+data SomeEncodable where
```

Comment:
I'm not super happy about these type names `SomeEncodable` and `SomeDecodable`. I'm open to suggestions.

Reply:
We can probably draw inspiration from here on how to do that: https://github.com/NorfairKing/autodocodec/issues/9

Reply:
The latest commits remove the need for these existential wrapper types by using `dimapCodec` instead.